optuna/storages/_journal/redis.py

```diff
@@ -0,0 +1,62 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import List
+
+from optuna._experimental import experimental_class
+from optuna._imports import try_import
+from optuna.storages._journal.base import BaseJournalLogStorage
+
+
+with try_import() as _imports:
+    import redis
+
+
+@experimental_class("3.1.0")
+class JournalRedisStorage(BaseJournalLogStorage):
+    """Redis storage class for Journal log backend.
+
+    Args:
+        url:
+            URL of the redis storage, password and db are optional. (ie: redis://localhost:6379)
+
+    """
+
+    def __init__(self, url: str) -> None:
+
+        _imports.check()
+
+        self._url = url
+        self._redis = redis.Redis.from_url(url)
+
+    def read_logs(self, log_number_from: int) -> List[Dict[str, Any]]:
+
+        self._redis.setnx("log_number", -1)
+        max_log_number_bytes = self._redis.get("log_number")
+        assert max_log_number_bytes is not None
+        max_log_number = int(max_log_number_bytes)
```

Comment:
It looks not intuitive for me that `read_logs` method inserts a record as a side-effect. How about returning empty array here?

```suggestion
        max_log_number_bytes = self._redis.get("log_number")
        if max_log_number_bytes is None:
            return []
        max_log_number = int(max_log_number_bytes)
```